packages/medusa/src/api-v2/store/payment-collections/[id]/payment-sessions/middlewares.ts

```diff
@@ -0,0 +1,11 @@
+import { transformBody } from "../../../../../api/middlewares"
+import { MiddlewareRoute } from "../../../../../loaders/helpers/routing/types"
+import { StorePostPaymentCollectionsPaymentSessionReq } from "./validators"
+
+export const storeCartRoutesMiddlewares: MiddlewareRoute[] = [
+  {
+    method: ["POST"],
+    matcher: "/store/payment-collections/:id/payment-sessions",
```

Comment:
comment: An alternative path could be `/store/payment-sessions`

Reply:
If we always know the collection ID (and we have to know it), then the existing endpoint makes sense. If you can operate on payment sessions without knowing the payment collection id (eg. you know the session ID and you want to delete it, but you don't know the collection ID), then what you proposed makes more sense.

I actually have a similar case in the product endpoints that I wanted to discuss, but we can do it once I open a PR